pkg/download/goget/goget.go

```diff
@@ -22,15 +22,24 @@ import (
 // New returns a download protocol by using
 // go get. You must have a modules supported
 // go binary for this to work.
-func New() download.Protocol {
-	return &goget{
-		goBinPath: env.GoBinPath(),
-		fs:        afero.NewOsFs(),
+func New() (download.Protocol, error) {
+	const op errors.Op = "goget.New"
+	goBin := env.GoBinPath()
+	fs := afero.NewOsFs()
+	mf, err := module.NewGoGetFetcher(goBin, fs)
+	if err != nil {
+		return nil, errors.E(op, err)
 	}
+	return &goget{
+		goBinPath: goBin,
+		fetcher:   mf,
+		fs:        fs,
+	}, nil
 }
 
 type goget struct {
 	goBinPath string
+	fetcher   module.Fetcher
```

Comment:
can you remove `goBinPath` and `fs` from the struct definition? They were being used to construct a module.Fetcher but now that it's being constructed just once, we can remove them.

Reply:
This comment ^ is not true, turns out we use both things in goget.list() -- my b